nano/secure/ledger_set_confirmed.cpp

```diff
@@ -0,0 +1,161 @@
+#include <nano/secure/ledger.hpp>
+#include <nano/secure/ledger_set_confirmed.hpp>
+#include <nano/store/account.hpp>
+#include <nano/store/component.hpp>
+#include <nano/store/confirmation_height.hpp>
+#include <nano/store/pending.hpp>
+#include <nano/store/pruned.hpp>
+
+nano::ledger_set_confirmed::ledger_set_confirmed (nano::ledger const & ledger) :
+	ledger{ ledger }
+{
+}
+
+std::optional<nano::uint128_t> nano::ledger_set_confirmed::balance (store::transaction const & transaction, nano::account const & account_a) const
+{
+	auto block = get (transaction, head (transaction, account_a));
+	if (!block)
+	{
+		return std::nullopt;
+	}
+	return block->balance ().number ();
+}
+
+// Balance for account containing hash
+std::optional<nano::uint128_t> nano::ledger_set_confirmed::balance (store::transaction const & transaction, nano::block_hash const & hash) const
+{
+	if (hash.is_zero ())
+	{
+		return std::nullopt;
+	}
+	auto block = get (transaction, hash);
+	if (!block)
+	{
+		return std::nullopt;
+	}
+	return block->balance ().number ();
+}
+
+bool nano::ledger_set_confirmed::exists (store::transaction const & transaction, nano::block_hash const & hash) const
```

Comment:
Code repetition. This can be implemented by calling `nano::ledger_set_confirmed::get`.